bitrise.yml

```diff
@@ -585,6 +597,50 @@ workflows:
                 exit 1
             fi
         is_always_run: true
+  Decision_to_run_UI_Tests:
+    steps:
+    - script@1:
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -e
+
+            echo "Commit on origin/main"
+            COMMIT_ON_MAIN_ORIGIN=`git log -n 1 origin/main --pretty=format:"%H"`
+            echo $COMMIT_ON_MAIN_ORIGIN
+
+            echo "Files changed with origin"
+            echo `git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_ORIGIN`
+
+            echo "Commit on main branch"
+            echo $GIT_CLONE_COMMIT_HASH
+            COMMIT_ON_MAIN_PR=`git log -n 2 main --pretty=format:"%H" | tail -n 1`
+            echo $COMMIT_ON_MAIN_PR
+
+            filesChanged=""
+            if [[ $COMMIT_ON_MAIN_PR == $COMMIT_ON_MAIN_ORIGIN ]]
+            then
+                echo "First commit, use origin/main"
+                filesChanged=`git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_ORIGIN`
+                echo $filesChanged
+            else
+                echo "Commits on PR"
+                echo "Files changed betwee commits"
+                echo `git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_PR`
+                filesChanged=`git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_PR`
+                echo $filesChanged
+            fi
+
+            doNotRunTests='^(\.[a-zA-Z]+\.y*|Docs/*|Sync*/*|.github/|[a-zA-Z]+\.md|[a-zA-Z]+\.sh|[a-zA-Z0-9]+[-_]+[a-zA-Z]+\.sh)|fastlane/*|taskcluster/*|L10nSnapshotTests/*|Shared/[a-zA-Z]+\.lproj/*|Shared/[A-Za-z]/*|WidgetKit/[a-zA-Z]+\.lproj/*|Client/*/[a-zA-Z]+\.lproj/*'
```

Comment:
🤔 Could we use CI's syntax to handle when the tests are run? I was thinking about this syntax from Github Actions.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters

Sorry I am not proficient in Bitrise's syntax yet. 🙇🏼‍♀️  Is it possible to use something like this? We can use the regular expression of the files' paths to denote the changes that UI tests are not needed. 💡 
https://devcenter.bitrise.io/en/builds/configuring-build-settings/selective-builds.html

Reply:
This is a script step on Bitrise, this means is like creating a shell script and that's the syntaxt. There is not any special syntax on Bitrise, in general it is a .yml fil, you have to follow that syntax, and then there are different type of steps, each require some options if they are pre-defined steps on bitrise or shell scripting if they are script steps.
This is the [bitrise documentation](https://devcenter.bitrise.io/en/steps-and-workflows/introduction-to-steps.html) about steps

> Sorry I am not proficient in Bitrise's syntax yet. 🙇🏼‍♀️ Is it possible to use something like this? We can use the regular expression of the files' paths to denote the changes that UI tests are not needed. 💡
> https://devcenter.bitrise.io/en/builds/configuring-build-settings/selective-builds.html

We can't use the default selective builds option Bitrise has, since it allows only the path to one file. The idea of that option is build or not apps depending on some changes if you have more than one app in the same repo for example